frontend/src/features/login/LoginCallbackForwardingPage.tsx

```diff
@@ -0,0 +1,93 @@
+import { useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import { Flex, Stack, Text } from '@chakra-ui/react'
+import { useFeatureValue } from '@growthbook/growthbook-react'
+
+import { LOGIN_ROUTE } from '~constants/routes'
+import Spinner from '~components/Spinner'
+
+import { useIsProxyIpCheck } from '~features/login/queries'
+
+export const ALLOWED_CALLBACK_FORWARDING_ROUTES = [
+  '/sgid/login',
+  '/mi/login',
+  '/api/v3/singpass/login',
+  '/api/v3/corppass/login',
+  '/api/v3/auth/sgid/login/callback',
+  '/api/v3/auth/sso/login/callback',
+]
+
+export const LoginCallbackForwardingPage = (): JSX.Element => {
+  const [params] = useSearchParams()
+
+  const route = params.get('route')
+  const forwarded = params.get('forwarded')
+
+  const callbackForwardingDefaultTimeout = useFeatureValue(
+    'callback-forwarding-default-timeout',
+    1000,
+  )
+  const callbackForwardingProxyTimeout = useFeatureValue(
+    'callback-forwarding-proxy-timeout',
+    10000,
+  )
```

Comment:
Might need to explain this portion a bit more as I think I might forget the nuances after 3 months 😆